taskvine/src/manager/vine_manager.c

```diff
@@ -4546,36 +4521,40 @@ int vine_submit(struct vine_manager *q, struct vine_task *t)
  * @param name The name of the library to be sent.
  * @return pointer to the library task if the operation succeeds, 0 otherwise.
  */
-
-static struct vine_task *send_library_to_worker(struct vine_manager *q, struct vine_worker_info *w, const char *name)
+struct vine_task *send_library_to_worker(struct vine_manager *q, struct vine_worker_info *w, const char *name)
 {
+	struct vine_task *library = vine_schedule_find_library(w, name);
```

Comment:
This assumes that there is exactly one library per worker, which isn't necessarily true.
e.g. an 8-core worker might run two instances of a 4-core library.

Reply:
I agree, but I want to have the basic case working of one library without any bugs. We need to rethink how libraries are assigned to workers. This is an emergency fix, if you will.